plugins/outputs/application_insights/application_insights.go

```diff
@@ -0,0 +1,410 @@
+			if err != nil {
+				return err
+			}
+
+			*telemetryProperty = metricValue
+
+			if usedFields != nil {
+				*usedFields = append(*usedFields, fieldName)
+			}
+
+			return nil
+		}
+	}
+
+	return fmt.Errorf("No field from the candidate list was found in the metric")
+}
+
+func setIntTelemetryProperty(
+	telemetryProperty *int,
+	candidateFields []string,
+	metric telegraf.Metric,
+	usedFields *[]string) error {
+
+	for _, fieldName := range candidateFields {
+		if metric.HasField(fieldName) {
+			metricValue, err := toInt(metric.Fields()[fieldName])
+			if err != nil {
+				return err
+			}
+
+			*telemetryProperty = metricValue
+
+			if usedFields != nil {
+				*usedFields = append(*usedFields, fieldName)
+			}
+
+			return nil
+		}
+	}
+
+	return fmt.Errorf("No field from the candidate list was found in the metric")
+}
+
+func addRemainingMeasurements(telemetry appinsights.Telemetry, metric telegraf.Metric, usedFields []string) {
+	properties := telemetry.GetProperties()
+
+	for fieldName, fieldValue := range metric.Fields() {
+		if contains(usedFields, fieldName) {
+			continue
+		}
+
+		float64Value, err := toFloat64(fieldValue)
+		if err != nil {
+			continue
+		}
+
+		// We would like to add the field to telemetry.Measurements here
+		// But because of issue https://github.com/Microsoft/ApplicationInsights-Go/issues/12
+		// this is currently impossible. So we are just going to add the measurement as a property instead.
+		// This is what we want to be able to do, eventually:
+		//     telemetry.GetMeasurements()[fieldName] = float64Value
+		// (plus handle name conflics)
+		addUniquelyNamedProperty(properties, fieldName, fmt.Sprintf("%g", float64Value))
+	}
+}
+
+func contains(set []string, val string) bool {
+	for _, elem := range set {
+		if elem == val {
+			return true
+		}
+	}
+
+	return false
+}
+
+func addUniquelyNamedProperty(properties map[string]string, name string, value string) {
+	if _, found := properties[name]; !found {
+		properties[name] = value
+	} else {
+		suffix := 1
+		for {
+			_, found = properties[name+"_"+strconv.Itoa(suffix)]
+			if !found {
+				break
+			}
+			suffix++
+		}
+		properties[name+"_"+strconv.Itoa(suffix)] = value
+	}
+}
+
+func toFloat64(value interface{}) (float64, error) {
+	switch v := value.(type) {
+	case int:
+		return float64(v), nil
+	case int8:
+		return float64(v), nil
+	case int16:
+		return float64(v), nil
+	case int32:
+		return float64(v), nil
+	case int64:
+		return float64(v), nil
+	case uint:
+		return float64(v), nil
+	case uint8:
+		return float64(v), nil
+	case uint16:
+		return float64(v), nil
+	case uint32:
+		return float64(v), nil
+	case uint64:
+		return float64(v), nil
+	case float32:
+		return float64(v), nil
+	case float64:
+		return v, nil
+	}
+
+	return 0.0, fmt.Errorf("[%s] cannot be converted to a float64 value", value)
+}
+
+func toInt(value interface{}) (int, error) {
+	if !is32BitChecked {
+		is32BitChecked = true
+		i := int(-1)
```

Comment:
What do you think of using sizeof to check 32bit? That is more straightforward to me.

Reply:
Sure, we can to that